.github/workflows/ci.yaml

```diff
@@ -47,29 +47,76 @@ jobs:
       - uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
         with:
           sharedKey: test
-      # The telephone-game example depends on ALSA, so we need this library to compile it
-      - name: Install libasound2-dev
-        run: sudo apt-get install -y libasound2-dev
-      - name: Remove root manifest
-        run: |
-          # HACK: The workspace smithy-rs generates is invalid.
-          # This can be removed once https://github.com/awslabs/smithy-rs/pull/957
-          # is merged into aws-sdk-rust.
-          rm Cargo.toml
       - name: Cargo Test
-        run: ./tools/ci/crate-range.py run ${{ matrix.crate_range }} -- cargo test --all-features
+        run: ./tools/ci/crate-range.py run ${{ matrix.crate_range }} --folder sdk -- cargo test --all-features
       - name: Cargo Doc
-        run: ./tools/ci/crate-range.py run ${{ matrix.crate_range }} -- cargo doc --all-features --no-deps
+        run: ./tools/ci/crate-range.py run ${{ matrix.crate_range }} --folder sdk -- cargo doc --all-features --no-deps
 
   # Psuedo-job that depends on the test job so that we don't have to enter
   # the myriad of test range combinations into GitHub's protected branch rules
-  require-tests:
-    needs: test
+  require-test-sdk:
+    needs: test-sdk
+    # Run this job even if its dependency jobs fail
+    if: always()
     runs-on: ubuntu-latest
-    name: Test Matrix Success
+    name: Test Matrix Success (SDK)
     steps:
-      - name: Run
-        run: echo "We should only get this far if the test matrix succeeded."
+      - name: Verify jobs succeeded
+        uses: re-actors/alls-green@3a2de129f0713010a71314c74e33c0e3ef90e696
+        with:
+          jobs: ${{ toJSON(needs) }}
+
+  generate-test-examples-matrix:
+    runs-on: ubuntu-latest
+    name: Generate Test Matrix (examples)
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate the matrix
+        id: generate-matrix
+        # Run the `crate-range.py` script to calculate a test matrix for the `test` job.
+        # This script outputs JSON that GitHub Actions can consume as a matrix definition.
+        # Rust versions to test against are arguments to this script.
+        run: echo "::set-output name=matrix::$(./tools/ci/crate-range.py generate-matrix -b ${BATCH_COUNT} --folder examples 1.54.0 1.56.1)"
```

Comment:
Just a note that when we merge the MSRV CI upgrade with this, it's not going to update this new one (or it will conflict). Not a problem, but we will need to keep an eye out for it.